### PR TITLE
Reimplements generic API calls. Moves the api base path to a higher l…

### PR DESCRIPTION
**Pull request by philals**
Moves the /api.xro/2.0/ part to the higher level.

Makes BaseAPIClient not abstract. So now it can be used as a generic client.

### CODE REVIEW

File: src/AccountingAPIClient.ts [hunk 1]
```diff
@@ -803,15 +805,15 @@ export class AccountingAPIClient extends BaseAPIClient {
...
+				const endpoint = this.accountingBasePath + 'trackingcategories/' + args.TrackingCategoryID + '/Options/' + args.TrackingOptionID;
 				return this.oauth1Client.delete<any>(endpoint);
 			},
 		}
 	};
 
 	public users = {
 		get: async (args?: { UserID?: string } & HeaderArgs & QueryArgs): Promise<UsersResponse> => {
-			let endpoint = 'users';
+			let endpoint = this.accountingBasePath + 'users';
```
Comment (dupski) [inline src/AccountingAPIClient.ts]: There is a hulluva lot of `this.accountingBasePath` in this file :)

Could I suggest passing in an (optional) basePath to the BaseAPIClient constructor instead?

**Comment by philals** [comment]
Cheers @dupski 

Changes made. I wanted to avoid passing an optional path into BaseAPI as then I think it makes it harder to use - and the object itself is less generic

Thoughts?

**Comment by dupski** [comment]
You could add the basePath option to the existing `ApiConfiguration` interface in the `BaseAPIClient` constructor so as not to require any new options, and users can ignore it if they want to set the full path themselves.

I am a big proponent of the DRY principle - having all those copy & pastes is a code smell. I'm not precious though - up to you @philals :)

**Comment by philals** [comment]
Cheers @dupski Have implemented your changes. Thinking about it.. All future APIs will have a basePath therefore having it implemented in our code like this makes it easier to add more.

Publishing once green.

**Pull request merged**